### PR TITLE
CLI Option to change max gas per block when selecting transactions

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
This PR introduces:

- `MaxBlockGasTransactionSelector`: A new class for transaction evaluation based on gas usage. It compares the gas used by a transaction with the maximum gas limit per block, set by the new CLI option introduced below

- A new CLI option `--plugin-linea-max-block-gas` for setting the maximum gas limit per block, enhancing user control over transaction selection.

### CODE REVIEW

File: acceptance-tests/src/test/java/linea/plugin/acc/test/BlockGasLimitTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    final String transactionHash1 =
+        sendTransactionWithGivenLengthPayload(Accounts.GENESIS_ACCOUNT_ONE_PRIVATE_KEY, web3j, 0);
+
+    final String largeGasTransactionHash =
+        sendTransactionWithGivenLengthPayload(
+            Accounts.GENESIS_ACCOUNT_TWO_PRIVATE_KEY, web3j, MAX_CALLDATA_SIZE);
+
+    final String transactionHash2 =
+        sendTransactionWithGivenLengthPayload(Accounts.GENESIS_ACCOUNT_ONE_PRIVATE_KEY, web3j, 0);
```
Comment (macfarla) [inline acceptance-tests/src/test/java/linea/plugin/acc/test/BlockGasLimitTest.java:56]: should we test a non-zero number other than max

File: arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockGasTransactionSelector.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      log.trace(
+          "Not selecting transaction, gas used {} greater than max block gas {}",
+          gasUsedByTransaction,
+          maxGasPerBlock);
+      return TransactionSelectionResult.invalid(TRANSACTION_GAS_EXCEEDS_MAX_BLOCK_GAS);
+    }
+
+    if (isTransactionExceedingMaxBlockGasLimit(gasUsedByTransaction)) {
+      log.trace(
+          "Not selecting transaction, cumulative gas used {} greater than max gas per block {}",
```
Comment (macfarla) [inline arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBlockGasTransactionSelector.java:50]: consistency of the message - `max gas per block` vs `max block gas` on line 42. I don't have a strong preference but should match the CLI option description

File: arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  private PluginTransactionSelector transactionSelector;
+
+  @BeforeEach
+  public void initialize() {
+    transactionSelector = new MaxBlockGasTransactionSelector(MAX_GAS_PER_BLOCK);
+  }
+
+  @Test
+  public void shouldSelectWhen_GasUsedByTransaction_IsLessThan_MaxGasUsedPerBlock() {
```
Comment (macfarla) [inline arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java:44]:
```suggestion
  public void shouldSelectWhen_GasUsedByTransaction_IsLessThan_MaxGasPerBlock() {
```

File: arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    var mockTransactionProcessingResult = mockTransactionProcessingResult(MAX_GAS_PER_BLOCK - 1);
+    verifyTransactionSelection(
+        transactionSelector,
+        mockTransaction(),
+        mockTransactionProcessingResult,
+        TransactionSelectionResult.SELECTED);
+  }
+
+  @Test
+  public void shouldSelectWhen_GasUsedByTransaction_IsEqual_MaxGasUsedPerBlock() {
```
Comment (macfarla) [inline arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java:54]:
```suggestion
  public void shouldSelectWhen_GasUsedByTransaction_IsEqual_MaxGasPerBlock() {
```

File: arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    var mockTransactionProcessingResult = mockTransactionProcessingResult(MAX_GAS_PER_BLOCK);
+    verifyTransactionSelection(
+        transactionSelector,
+        mockTransaction(),
+        mockTransactionProcessingResult,
+        TransactionSelectionResult.SELECTED);
+  }
+
+  @Test
+  public void shouldNotSelectWhen_GasUsedByTransaction_IsGreaterThan_MaxGasUsedPerBlock() {
```
Comment (macfarla) [inline arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java:64]:
```suggestion
  public void shouldNotSelectWhen_GasUsedByTransaction_IsGreaterThan_MaxGasPerBlock() {
```

File: arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    var mockTransactionProcessingResult = mockTransactionProcessingResult(MAX_GAS_PER_BLOCK + 1);
+    verifyTransactionSelection(
+        transactionSelector,
+        mockTransaction(),
+        mockTransactionProcessingResult,
+        TransactionSelectionResult.invalid(TRANSACTION_GAS_EXCEEDS_MAX_BLOCK_GAS));
+  }
+
+  @Test
+  public void shouldNotSelectWhen_CumulativeGasUsed_IsGreaterThan_MaxGasUsedPerBlock() {
```
Comment (macfarla) [inline arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/MaxBlockGasTransactionSelectorTest.java:74]:
```suggestion
  public void shouldNotSelectWhen_CumulativeGasUsed_IsGreaterThan_MaxGasPerBlock() {
```

File: arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/LineaTransactionSelectorCliOptions.java [hunk 7]
```diff
@@ -22,9 +22,10 @@
 public class LineaTransactionSelectorCliOptions {
   public static final int DEFAULT_MAX_BLOCK_CALLDATA_SIZE = 70000;
   private static final String DEFAULT_MODULE_LIMIT_FILE_PATH = "moduleLimitFile.json";
-
+  public static final long DEFAULT_MAX_GAS_PER_BLOCK = Long.MAX_VALUE;
```
Comment (fab-10) [inline arithmetization/src/main/java/net/consensys/linea/sequencer/txselection/LineaTransactionSelectorCliOptions.java:25]: could we put here the expected default for Linea? If I am not wrong it should be 30M

**Review by fab-10 — APPROVED** [review]
LGTM, just a suggestion about the default value

**Pull request merged**